Makefile

```
@@ -180,22 +181,31 @@ cf-deploy: ## Deploys the app to Cloud Foundry
 	$(if ${CF_SPACE},,$(error Must specify CF_SPACE))
 	@cf app --guid notify-admin || exit 1
 	# cancel any existing deploys to ensure we can apply manifest (if a deploy is in progress you'll see ScaleDisabledDuringDeployment)
-	cf v3-cancel-zdt-push ${CF_APP} || true
+	cf cancel-deployment ${CF_APP} || true
 
-	cf v3-apply-manifest ${CF_APP} -f <(make -s generate-manifest)
-	CF_STARTUP_TIMEOUT=10 cf v3-zdt-push ${CF_APP} --wait-for-deploy-complete  # fails after 5 mins if deploy doesn't work
+	# generate manifest (including secrets) and write it to CF_MANIFEST_PATH (in /tmp/)
+	make -s CF_APP=${CF_APP} generate-manifest > ${CF_MANIFEST_PATH}
+	# reads manifest from CF_MANIFEST_PATH
+	CF_STARTUP_TIMEOUT=10 cf push ${CF_APP} --strategy=rolling -f ${CF_MANIFEST_PATH}
+	# delete old manifest file
+	rm ${CF_MANIFEST_PATH}
 
 .PHONY: cf-deploy-prototype
 cf-deploy-prototype: cf-target ## Deploys the first prototype to Cloud Foundry
-	cf push -f <(CF_APP=notify-admin-prototype make -s generate-manifest)
+	make -s CF_APP=notify-admin-prototype generate-manifest > ${CF_MANIFEST_PATH}
+	cf push notify-admin-prototype --strategy=rolling -f ${CF_MANIFEST_PATH}
+	rm ${CF_MANIFEST_PATH}
 
 .PHONY: cf-deploy-prototype-2
 cf-deploy-prototype-2: cf-target ## Deploys the second prototype to Cloud Foundry
-	cf push -f <(CF_APP=notify-admin-prototype-2 make -s generate-manifest)
+	make -s CF_APP=notify-admin-prototype-2 generate-manifest > ${CF_MANIFEST_PATH}
+	cf push notify-admin-prototype-2 --strategy=rolling -f ${CF_MANIFEST_PATH}
+	rm ${CF_MANIFEST_PATH}
 
 .PHONY: cf-rollback
 cf-rollback: cf-target ## Rollbacks the app to the previous release
-	cf v3-cancel-zdt-push ${CF_APP}
+	cf cancel-deployment ${CF_APP}
+	rm ${CF_MANIFEST_PATH}
```

should have realised in the API pull request, my bad. i think this rm command should have the `-f` arg. That way, if the file doesn't exist, there's no output and the makefile doesn't fail. It's most relevant to this rollback command, but can't hurt in the other commands I think. If you agree I can update the api makefile

```suggestion
	rm -f ${CF_MANIFEST_PATH}
```

---

Agreed

---

I've squashed and pushed up changes